### PR TITLE
some anyparameter labels for classifier algos

### PR DESCRIPTION
**Pull request by gf712**

### CODE REVIEW

File: src/shogun/classifier/LDA.cpp [hunk 1]
```diff
@@ -57,11 +57,13 @@ void CLDA::init()
...
-	SG_ADD(&m_bdc_svd, "bdc_svd", "Use BDC-SVD algorithm");
+	SG_ADD(
+	    &m_bdc_svd, "bdc_svd", "Use BDC-SVD algorithm",
+	    ParameterProperties::SETTING)
 	SG_ADD_OPTIONS(
-	    (machine_int_t*)&m_method, "method",
-	    "Method used for LDA calculation", ParameterProperties::NONE,
-	    SG_OPTIONS(AUTO_LDA, SVD_LDA, FLD_LDA));
+	    (machine_int_t*)&m_method, "method", "Method used for LDA calculation",
+	    ParameterProperties::SETTING,
```
Comment (gf712) [inline src/shogun/classifier/LDA.cpp:65]: @karlnapf HYPER or SETTING?

File: src/shogun/classifier/PluginEstimate.cpp [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
+	    &m_neg_pseudo, "neg_pseudo", "pseudo count for negative class",
+	    ParameterProperties::MODEL);
 	SG_ADD(
-	    &pos_model, "pos_model", "LinearHMM modelling positive class.");
+	    &pos_model, "pos_model", "LinearHMM modelling positive class.",
+	    ParameterProperties::MODEL);
 	SG_ADD(
-	    &neg_model, "neg_model", "LinearHMM modelling negative class.");
+	    &neg_model, "neg_model", "LinearHMM modelling negative class.",
+	    ParameterProperties::MODEL);
```
Comment (gf712) [inline src/shogun/classifier/PluginEstimate.cpp:34]: assuming these should all be MODEL?

File: src/shogun/classifier/mkl/MKL.cpp [hunk 3]
```diff
@@ -271,23 +271,25 @@ void CMKL::register_params()
...
 	m_parameters->add_vector(&beta_local, &beta_local_size, "beta_local", "subkernel weights on L1 term of elastic net mkl");
-	watch_param("beta_local", &beta_local, &beta_local_size);
+	watch_param("beta_local", &beta_local, &beta_local_size,
+			AnyParameterProperties("subkernel weights on L1 term of elastic net mkl",
+					ParameterProperties::MODEL));
 
-	SG_ADD(&mkl_iterations, "mkl_iterations", "number of mkl steps");
-	SG_ADD(&mkl_epsilon, "mkl_epsilon", "mkl epsilon");
+	SG_ADD(&mkl_iterations, "mkl_iterations", "number of mkl steps", ParameterProperties::MODEL);
+	SG_ADD(&mkl_epsilon, "mkl_epsilon", "mkl epsilon", ParameterProperties::HYPER);
```
Comment (gf712) [inline src/shogun/classifier/mkl/MKL.cpp:288]: @karlnapf epsilon values: HYPER or SETTING?

File: src/shogun/classifier/svm/OnlineSVMSGD.cpp [hunk 4]
```diff
@@ -209,14 +209,14 @@ void COnlineSVMSGD::init()
 	SG_ADD(&C1, "C1", "Cost constant 1.", ParameterProperties::HYPER);
 	SG_ADD(&C2, "C2", "Cost constant 2.", ParameterProperties::HYPER);
 	SG_ADD(&lambda, "lambda", "Regularization parameter.", ParameterProperties::HYPER);
-	SG_ADD(&wscale, "wscale", "W scale");
-	SG_ADD(&bscale, "bscale", "b scale");
-	SG_ADD(&epochs, "epochs", "epochs");
+	SG_ADD(&wscale, "wscale", "W scale", ParameterProperties::HYPER);
+	SG_ADD(&bscale, "bscale", "b scale", ParameterProperties::HYPER);
+	SG_ADD(&epochs, "epochs", "epochs", ParameterProperties::HYPER);
```
Comment (gf712) [inline src/shogun/classifier/svm/OnlineSVMSGD.cpp:214]: I couldn't tell what this epochs parameter does? is that the number of epochs to run? or is it the number of epochs as online training evolves?

File: src/shogun/classifier/svm/SVM.cpp [hunk 5]
```diff
@@ -300,7 +300,7 @@ float64_t* CSVM::get_linear_term_array()
 	return a;
 }
 
-void CSVM::set_linear_term(const SGVector<float64_t> linear_term)
+void CSVM::set_linear_term(const SGVector<float64_t>& linear_term)
```
Comment (gf712) [inline src/shogun/classifier/svm/SVM.cpp:303]: unrelated, just thought this should be ref

File: src/shogun/classifier/svm/SVMOcas.cpp [hunk 6]
```diff
@@ -347,12 +347,12 @@ void CSVMOcas::init()
...
-	SG_ADD(&use_bias, "use_bias", "Indicates if bias is used.");
-	SG_ADD(&epsilon, "epsilon", "Convergence precision.");
-	SG_ADD(&bufsize, "bufsize", "Maximum number of cutting planes.");
+	SG_ADD(&use_bias, "use_bias", "Indicates if bias is used.", ParameterProperties::SETTING);
+	SG_ADD(&epsilon, "epsilon", "Convergence precision.", ParameterProperties::HYPER);
+	SG_ADD(&bufsize, "bufsize", "Maximum number of cutting planes.", ParameterProperties::HYPER);
 	SG_ADD_OPTIONS(
-	    (machine_int_t*)&method, "method", "SVMOcas solver type.",
-	    ParameterProperties::NONE, SG_OPTIONS(SVM_OCAS, SVM_BMRM));
+	    (machine_int_t*)&method, "method", "SVMOcas solver type.", ParameterProperties::SETTING,
```
Comment (gf712) [inline src/shogun/classifier/svm/SVMOcas.cpp:354]: @karlnapf SETTING or HYPER?

File: src/shogun/classifier/LDA.cpp (hunk 1 above)
Comment (karlnapf) [reply src/shogun/classifier/LDA.cpp:65]: Setting!

File: src/shogun/classifier/PluginEstimate.cpp (hunk 2 above)
Comment (karlnapf) [reply src/shogun/classifier/PluginEstimate.cpp:34]: I think this class doesn’t work anyways. But yes model

File: src/shogun/classifier/mkl/MKL.cpp (hunk 3 above)
Comment (karlnapf) [reply src/shogun/classifier/mkl/MKL.cpp:288]: I would make those setting

Comment (karlnapf) [reply src/shogun/classifier/mkl/MKL.cpp:288]: Or hyper, but not model

File: src/shogun/classifier/svm/SVMOcas.cpp (hunk 6 above)
Comment (karlnapf) [reply src/shogun/classifier/svm/SVMOcas.cpp:354]: Setting

File: src/shogun/classifier/svm/OnlineSVMSGD.cpp (hunk 4 above)
Comment (karlnapf) [reply src/shogun/classifier/svm/OnlineSVMSGD.cpp:214]: Think latter so hyper is good

**Comment by karlnapf** [comment]
updates here?

**Comment by gf712** [comment]
hey, I have been on annual leave. I will try and push something today :)

**Comment by gf712** [comment]
@karlnapf the CI error is due to a bug in the test. When the object is cloned in cross validation it does not copy the set solver type and defaults to L2R_L1LOSS_SVC_DUAL (instead of L2R_L2LOSS_SVC). That's because this was originally set to ParameterProperties::NONE and I changed it to SETTING

**Comment by karlnapf** [comment]
Looks good. Merge?

**Comment by gf712** [comment]
Yup, I think so!

**Pull request merged**